solutions/problem2.js

```diff
@@ -0,0 +1,9 @@
+const { isArrayEmpty } = require('./utils')
+
+const attempt = (promises, last = 0) => isArrayEmpty(promises)
+  ? Promise.reject(last)
+  : promises
+      .shift()()
```

Comment:
Jedyne co mi się tu nie podoba to ten shift - osoba używająca funkcji może się nie spodziewać, że modyfikuje ona arraya. Z mojego doświadczenia najtrudniejsze do debugowania błędy to właśnie przypadkowe modyfikacje list. ;)

Zamiast tego można by użyć destrukturyzacji z es6 - `[first, ...rest]`.